tests/ticket/search.rb

```diff
@@ -108,6 +285,399 @@
+    end
+
+    it 'should success if the owners are valid' do
+        ownerId1 = 8
+        ownerId2 = 1
+        ownerId3 = 2
+
+        ownerIdList1 = [ownerId1].to_json
+        ownerIdList2 = [ownerId2, ownerId3, ownerId1].to_json
+        ownerIdList3 = [ownerId3, ownerId1].to_json
+
+        ownerId1 = ownerId1.to_s
+        ownerId2 = ownerId2.to_s
+        ownerId3 = ownerId3.to_s
+
+        for page in $pages
+            result = request('/ticket/search', {
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                owners: ownerIdList1
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                (ticket['owner']['id']).should.equal(ownerId1)
+            end
+
+            result = request('/ticket/search', {
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                owners: ownerIdList2
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                ticketOwnerId = ticket['owner']['id']
+                if ticketOwnerId === ownerId1
+                    ticketOwnerId.should.equal(ownerId1)
+                elsif ticketOwnerId === ownerId2
+                    ticketOwnerId.should.equal(ownerId2)
+                elsif ticketOwnerId === ownerId3
+                    ticketOwnerId.should.equal(ownerId3)
+                else
+                    1.should.equal(0)
+                end
+            end
+
+            result = request('/ticket/search', {
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                owners: ownerIdList3
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                ticketOwnerId = ticket['owner']['id']
+                if ticketOwnerId === ownerId1
+                    ticketOwnerId.should.equal(ownerId1)
+                elsif ticketOwnerId === ownerId3
+                    ticketOwnerId.should.equal(ownerId3)
+                else
+                    1.should.equal(0)
+                end
+            end
+        end
+    end
+
+    it 'should success if the closed value is valid' do
+        for page in $pages
+            result = request('/ticket/search', {
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                closed: 1
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                (ticket['closed']).should.equal(true)
+            end
```

Comment:
Again, not only check if the returned tickets are all closed, but also if all the tickets that are closed in the corresponding page are the ones that are returned (at least that the size matches).

Reply:
I'd avoid checking sizes; other tests can create tickets and it will affect these